File: src/main/java/io/gaegul/ch01/RedheadDuck.java
package io.gaegul.ch01;

import io.gaegul.ch01.behavior.FlyWithWings;
import io.gaegul.ch01.behavior.Quack;

public class RedheadDuck extends Duck {

	public RedheadDuck() {
		super(new FlyWithWings(), new Quack());
	}

	@Override
	public void display() {
		System.out.println("흰죽지오리");
	}
}

File: src/main/kotlin/io/gaegul/ch01/RedheadDuck.kt
package io.gaegul.ch01

import io.gaegul.ch01.behavior.FlyWithWings
import io.gaegul.ch01.behavior.Quack

class RedheadDuck: Duck(
    Quack(), FlyWithWings()
) {

    override fun display() {
        println("흰죽지오리")
    }
}